person.rb

```diff
@@ -0,0 +1,31 @@
+require_relative 'nameable'
+class Person < Nameable
+  # accessing the attributes by attr_accessor method
+  attr_accessor :name, :age
+  attr_reader :id
+
+  def initialize(age, name = 'Unknown', parent_permission: true)
+    # instance variables
+    @id = Random.rand(1..300)
+    @name = name
+    @age = age
+    @parent_permission = parent_permission
+    super
+  end
+
```

Comment:
- [ ] Kindly make sure you update the `super` method to this `super()` as right now when trying to create a new `Person` it raises an `ArgumentError` and **(OPTIONAL)** for best practices, I can suggest you move it just before the `@id` instance variable(once you have updated it)